modules/@apostrophecms/ai-helper-rich-text-widget/index.js

```diff
@@ -0,0 +1,88 @@
+const marked = require('marked');
+
+// Canned sample results for cheap & offline testing
+const mockResults = {
+  "id": "cmpl-76iWL6JGwp0S3MoFx77100hd4cZjr",
+  "object": "text_completion",
+  "created": 1681835461,
+  "model": "text-davinci-003",
+  "choices": [
+    {
+      "text": "\n# Glamping Cats\n\n## The Feline Appeal\nGlamping cats have become increasingly popular in recent years, with many people choosing to include their cats in their glamping experience. Cats are incredibly adaptive and can quickly become accustomed to new environments with minimal stress. They can also make perfect glamping companions, as their intelligence and inquisitive nature ensure that they will quickly find ways to entertain themselves and those around them.\n\n## Glamping Must-Haves\nBefore taking your cat on a glamping vacation, make sure you are prepared. Consider bringing items such as litterbox, food dish, bedding material, and toys. Each of these items can help create an enjoyable and stress-free environment for your cat while glamping.\n\n## RV Luxuries\nRVs come equipped with a variety of amenities that make glamping with cats more enjoyable. Consider portable kennels and carriers for your cat, as well as waterproof mats and blankets for lounging. RV heating capabilities can also create a comfortable and safe environment for your cat, no matter the temperature outside. \n\n## All-Weather Fun\nGlamping with cats can provide opportunities for outdoor fun regardless of the weather. Rainy days can be spent cuddling cats inside, while sunny days can offer the adventure of roaming the outdoors. Glamping can also offer cats the opportunity to meet new people and explore new environments.\n\n## A Special Bond\nUltimately, glamping with cats can create a special bond between you and your pet. Cats tend to be incredibly social creatures and thrive off of human interaction. Glamping with cats offers the perfect opportunity to strengthen your relationship and appreciate the wonders of nature.",
+      "index": 0,
+      "logprobs": null,
+      "finish_reason": "stop"
+    }
+  ],
+  "usage": {
+    "prompt_tokens": 32,
+    "completion_tokens": 342,
+    "total_tokens": 374
+  }
+};
+
+module.exports = {
+  improve: '@apostrophecms/rich-text-widget',
+  beforeSuperClass(self) {
+    self.options.editorInsertMenu = {
+      ...self.options.editorInsertMenu,
+      ai: {
+        icon: 'robot-icon',
+        label: 'aposAiHelper:generateTextLabel',
+        component: 'AposAiHelperTextDialog',
+        description: 'aposAiHelper:generateTextDescription'
+      }
```

Comment:
Would be good to document it:

> add `insert: [ 'ai' ]` to any rich text widget to add this AI feature in the Rich Text Widget

Reply:
Once this option added, it works very well!

| Header | Header |
|--------|--------|
| ![image](https://user-images.githubusercontent.com/8301962/234027236-08333f7d-ce0f-43ae-8567-1734652b89d9.png) | ![image](https://user-images.githubusercontent.com/8301962/234027533-66a01f38-68a6-4df5-bd6b-e80ed5e8bb56.png) |

Reply:
Thank you for figuring that out! I documented it.